### PR TITLE
boards: introduce bytesatwork byteSENSI-L

### PR DESCRIPTION
**Pull request by morandg**
The byteSENSI-L is a fun LoRa device based on nRF52 MCU that integrates many sensors.

### CODE REVIEW

**Comment by morandg** [comment]
Hi Zephyr maintainers!

I understand you have a lot of PR to look at and sorry for asking... Would it be possible to have a first feedback about this? We would be very thankful if this could be merged before the release 3.7.0.

Thanks!

File: boards/bytesatwork/bytesensi_l/board.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+	/* Enable 1-wire to fix i2c bus */
+	const struct gpio_dt_spec enable_1_wire_gpio =
+		GPIO_DT_SPEC_GET_OR(DT_PATH(zephyr_user), enable_1wire_gpios, {0});
+
+	if (!gpio_is_ready_dt(&enable_1_wire_gpio)) {
+		return -ENODEV;
+	}
+
+	return gpio_pin_configure_dt(&enable_1_wire_gpio, GPIO_OUTPUT_ACTIVE);
```
Comment (anangl) [inline boards/bytesatwork/bytesensi_l/board.c:20]: Why not use `gpio-hog` for this? See for example 12cca8939e3d4eb7143d0a6086dcf600694bbe8d.

**Review by anangl — COMMENTED** [review]
Looks good. Just one improvement suggestion.

**Review by gmarull — CHANGES_REQUESTED** [review]
use gpio-hogs for 1wire gpio

File: boards/bytesatwork/bytesensi_l/board.c (hunk 1 above)
Comment (morandg) [reply boards/bytesatwork/bytesensi_l/board.c]: Awesome, I learned a new trick today, thanks!

File: boards/bytesatwork/bytesensi_l/bytesensi_l.dts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+	zephyr,user {
+		enable-1wire-gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+	};
```
Comment (anangl) [inline boards/bytesatwork/bytesensi_l/bytesensi_l.dts:27]: This is no longer needed.

Comment (morandg) [reply boards/bytesatwork/bytesensi_l/bytesensi_l.dts]: Yes, I thought about that during the lunch :see_no_evil: , good catch!

**Review by gmarull — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

File: boards/bytesatwork/bytesensi_l/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+#. Build a :ref:`bluetooth-samples`:
+
+   Most of the bluetooth samples should work out of the box, for example,
+   building the :ref:`peripheral_hids` sample:
+
+   .. zephyr-app-commands::
+      :zephyr-app: samples/bluetooth/peripheral_hids
+      :board: bytesensi_l
+      :goals: build flash
+      :compact:
```
Comment (kartben) [inline boards/bytesatwork/bytesensi_l/doc/index.rst:208]: Please only list one, there is no point in enumerating so many sample applications

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/bytesatwork/bytesensi_l/doc/index.rst (hunk 3 above)
Comment (morandg) [reply boards/bytesatwork/bytesensi_l/doc/index.rst:208]: Alright, I'll keep only the ``hello world`` sample, thanks!

File: MAINTAINERS.yml [hunk 4]
```diff
@@ -526,6 +526,14 @@ Board/SoC configuration:
   labels:
     - "area: Board/SoC configuration"
 
+bytesatwork Platforms:
+  status: maintained
+  maintainers:
+    - morandg
```
[the remark this answers is not in the document]
Comment (morandg) [reply MAINTAINERS.yml:532]: Thanks for your feedback! The reason to fillout the ```MAINTAINERS.yml``` was to pass the build workflow
* Will it be merged if I revert this change?
* What should I do to fix the build without adding us as maintainers?

> Can you please create a contributor nomination issue for each of you?

That would be for me:
* https://github.com/zephyrproject-rtos/zephyr/issues/73666

[the remark this answers is not in the document]
Comment (aescolar) [reply MAINTAINERS.yml]: @morandg if the only CI failure is the orphan check for this board files we can bypass it and merge still.

**Comment by dariis** [comment]
And for me it would be: #73706

File: boards/bytesatwork/bytesensi_l/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+The byteSENSI-L board can be flashed with the SEGGER JLink programmer.
+You can build the following samples as demo application:
+
+#. Build the :ref:`hello_world` sample:
```
Comment (kartben) [inline boards/bytesatwork/bytesensi_l/doc/index.rst:135]:
```suggestion
You can build and flash applications in the usual way. Here is an example for
the :ref:`hello_world` application.
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — APPROVED** [review]
LGTM from a documentation standpoint

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
Very nice.

**Comment by morandg** [comment]
Fantastic, thanks everyone!

**Pull request merged**